### PR TITLE
[search] A probabilistic tweak to locality scorer.

### PR DESCRIPTION
**Pull request by mpimenov**
https://jira.mail.ru/browse/MAPSME-1380

### CODE REVIEW

File: search/geocoder.cpp [hunk 1]
```diff
@@ -694,9 +694,14 @@ void Geocoder::FillLocalityCandidates(coding::CompressedBitVector const * filter
   for (size_t startToken = 0; startToken < m_numTokens; ++startToken)
   {
     CBVPtr intersection;
+    CBVPtr unfilteredIntersection;
```
Comment (ygorshenin) [inline search/geocoder.cpp]: I guess unfilteredIntersection must be set to full unconditionally.

Comment (mpimenov) [inline search/geocoder.cpp]: All uses of ```unfilteredIntersection``` are guarded. But setting to full is cheap, so

done

**Comment by mpimenov** [comment]
PTAL

**Comment by mpimenov** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**